istioctl/pkg/waypoint/waypoint.go

```diff
@@ -49,33 +50,63 @@ var (
 
 	deleteAll bool
 
-	trafficType       string
+	trafficType       = constants.ServiceTraffic
 	validTrafficTypes = sets.New(constants.ServiceTraffic, constants.WorkloadTraffic, constants.AllTraffic, constants.NoTraffic)
 
-	waypointName string
+	waypointName      = constants.DefaultNamespaceWaypoint
+	enrolledNamespace = constants.DefaultEnrolledNamespace
 )
 
 const waitTimeout = 90 * time.Second
 
 func Cmd(ctx cli.Context) *cobra.Command {
-	makeGatewayName := func(name string) string {
-		if name == "" {
-			name = constants.DefaultNamespaceWaypoint
+	annotateNamespaceWithWaypoint := func(kubeClient kube.CLIClient) error {
+		nsObj, err := kubeClient.Kube().CoreV1().Namespaces().Get(context.Background(), enrolledNamespace, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("namespace: %s not found", enrolledNamespace)
+		} else if err != nil {
+			return fmt.Errorf("failed to get namespace %s: %v", enrolledNamespace, err)
 		}
-		return name
+		if nsObj.Annotations == nil {
+			nsObj.Annotations = map[string]string{}
+		} else if _, exists := nsObj.Annotations[constants.AmbientUseWaypoint]; exists && enrolledNamespace == constants.DefaultEnrolledNamespace {
+			// If the namespace is already annotated with a Waypoint name and the user did not specify a Waypoint name, return early.
+			return nil
+		}
+		nsObj.Annotations[constants.AmbientUseWaypoint] = waypointName
+		return nil
```

Comment:
It seems like we don't kclient.Update the namespace after we mutate the object
```suggestion
		nsObj.Annotations[constants.AmbientUseWaypoint] = waypointName
		// need to update the namespace with kube api here
		return nil
```